politeiawww/pi.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	piplugin "github.com/decred/politeia/plugins/pi"
+	pi "github.com/decred/politeia/politeiawww/api/pi/v1"
+	"github.com/decred/politeia/politeiawww/user"
+)
+
+func convertPropStateFromPi(s pi.PropStateT) piplugin.PropStateT {
+	switch s {
+	case pi.PropStateUnvetted:
+		return piplugin.PropStateUnvetted
+	case pi.PropStateVetted:
+		return piplugin.PropStateVetted
+	}
+	return piplugin.PropStateInvalid
+}
+
+// newComment calls the pi plugin to add new comment.
+func (p *politeiawww) piCommentNew(cn pi.CommentNew, usr *user.User) (*piplugin.CommentNewReply, error) {
```

Comment:
This is kinda bad form.  You shouldn't assume what the calling function is, meaning, by passing in the `pi.CommentNew` you're making the assumption that the calling function is the pi `CommentNew` route?  What if we want to call this function somewhere else?  You should break out the arguments into what the function actually needs (token, parentID, etc) so that you're not limiting where this can be called.

Also, if the only field you need from the user is the user ID then just pass in that single field.  It makes it more explicit what exactly is being used.

Reply:
Makes sense, updating